module3/lessons/Week4/UpdatingResources.md

```diff
@@ -0,0 +1,219 @@
+---
+layout: page
+title: Updating Resources in MVC
+---
+
+## Learning Goals
+- Use a form to update single resources
+- Use a form to create a one-to-many relationship
+
+During today's lesson, we are going to be adding the ability to edit our Movie and Song records using forms.
+
+## Warm Up
+
+<section class='call-to-action' markdown='1'>
+
+Before we jump into coding, let's brainstorm what the HTTP Request/Response Cycle will look like for editing a Movie.  In small groups, create a diagram that outlines the requests and responses that will allow us to edit an existing resource.  Below are some resources that will help you create this diagram!
+* [How the Web Works](/module3/lessons/week1/HowTheWebWorks)
+* [ReSTful Routes]() TODO: Get ReSTful routes Lesson link
+* [Creating a Resource]() TODO: Get Creating a Resource Lesson link
+
+Be ready to share out!
+
+</section>
+
+## TDD Updating single resource
+
+Updating a resource will look a lot like creating a resource.  We will need an action that returns a view with a form, and an action that makes database changes based on the form submission.  The biggest difference is that we are going to be associating an _existing_ record with the form; so our form-view will need additional information.
+
+<section class='call-to-action' markdown='1'>
+
+**With a Partner**  
+Add a test to our MoviesControllerTests that satisfies this user-story
+```
+As a User
+When I visit '/movies/1/edit'
+Than I see a form to edit the movie
+    And I see that the Title and Genre for that movie are pre-populated in the form.
+```
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### One Solution
+
+```c#
+[Fact]
+public async Task Edit_ReturnsFormViewPrePopulated()
+{
+    // Arrange
+    var context = GetDbContext();
+    var client = _factory.CreateClient();
+
+    Movie movie = new Movie { Title = "Spaceballs", Genre = "Comedy" };
+    context.Movies.Add(movie);
+    context.SaveChanges();
+
+    // Act
+    var response = await client.GetAsync($"/movies/{movie.Id}/edit");
+    var html = await response.Content.ReadAsStringAsync();
+
+    // Assert
+    Assert.Contains("Edit Movie", html);
+    Assert.Contains(movie.Title, html);
+    Assert.Contains(movie.Genre, html);
+}
+```
+
+</section>
+
+**Code-Along**  
+Now let's build the action that will make this test pass!
+
+### HTTP and REsTful Routes
+
+You will note in our walkthrough that even though the most **ReST**ful way to send an update request is with a `PUT` method; but, we are using `POST` in our form - why?
```

Comment:
This is a super small thing, but I have never seen this capitalization, I'm only every seen RESTful or restful 🤷‍♀️

Reply:
I use this capitalization because there is no E in the acronym.  **R**epresentational **S**tate **T**ransfer.  I guess I am the source that I'm pulling from because I hate acronyms 🙃.  If you feel strongly that we should use `REST`, I can live with it.